src/node-jspm-jasmine.js

```diff
@@ -141,17 +180,73 @@ export function runTests(opts, errCallback = function() {}) {
 							importTheseTestFiles();
 						}
 					})
-					.catch(errCallback);
+					.catch((ex) => {
+						if (opts.coverage && opts.coverage.tempDirectory) {
+							// remove temporary directory
+							rimraf.sync(opts.coverage.tempDirectory)
+						}
+						errCallback(ex)
+					});
 				});
 			});
 		}
 	} catch(ex) {
-		removeTempFiles();
+		if (opts.coverage && opts.coverage.tempDirectory) {
+			// remove temporary directory
+			rimraf.sync(opts.coverage.tempDirectory)
+		}
 		errCallback(new Error(`Jasmine or Jspm may not be properly configured -- '${ex.toString()}'`));
 	}
 }
 
 function importTestFiles(SystemJS, jasmine, specDir, specFiles, coverage, errCallback) {
+
+	if (coverage) {
+		// the "onComplete" hook to prevent jasmine's self-righteous exit
+		jasmine.onComplete(function (passed) {
+			// avoid misalignment with jasmine's output
+			console.log('');
+			// import the rest of the modules not already imported (evaluated)
+			// as dependencies of specs
+			Promise.all(Object.keys(coverage.coverageFiles).map(function (file) {
+				return SystemJS.import(path.join(process.cwd(), file));
+			})).then(function () {
+				if (typeof __coverage__ === 'undefined') {
+					console.log("No coverage was collected for files matching globs " + coverage.files);
+				} else {
+					try {
+						const collector = remap(__coverage__)
+						let report = Report.create(
+							(coverage.reporter || 'html'),
+							{
+								dir: coverage.dir
+							}
+						);
+						report.writeReport(collector, true);
```

Comment:
I found out where the error is being thrown -- it is at https://github.com/gotwarlost/istanbul/blob/master/lib/report/html.js#L210.

My guess is that it might be related to the last comment in [this jspm issue](https://github.com/jspm/jspm-cli/issues/1804#issuecomment-220105766). Alternatively, it might be related to us trying to generate a coverage report for a file that doesn't have sourcemaps when it comes through SystemJS.instantiate, or something like that